pandas/core/groupby/generic.py

```diff
@@ -1309,42 +1309,51 @@ def _transform_general(self, func, *args, **kwargs):
         gen = self.grouper.get_iterator(obj, axis=self.axis)
         fast_path, slow_path = self._define_paths(func, *args, **kwargs)
 
-        for name, group in gen:
-            object.__setattr__(group, "name", name)
-
-            # Try slow path and fast path.
-            try:
-                path, res = self._choose_path(fast_path, slow_path, group)
-            except TypeError:
-                return self._transform_item_by_item(obj, fast_path)
-            except ValueError as err:
-                msg = "transform must return a scalar value for each group"
-                raise ValueError(msg) from err
-
+        def process_result(applied, group, res):
             if isinstance(res, Series):
 
                 # we need to broadcast across the
                 # other dimension; this will preserve dtypes
                 # GH14457
                 if not np.prod(group.shape):
-                    continue
-                elif res.index.is_(obj.index):
-                    r = concat([res] * len(group.columns), axis=1)
-                    r.columns = group.columns
-                    r.index = group.index
+                    pass
                 else:
-                    r = self.obj._constructor(
-                        np.concatenate([res.values] * len(group.index)).reshape(
-                            group.shape
-                        ),
-                        columns=group.columns,
-                        index=group.index,
-                    )
-
-                applied.append(r)
+                    if res.index.is_(obj.index):
+                        r = concat([res] * len(group.columns), axis=1)
+                        r.columns = group.columns
+                        r.index = group.index
+                    else:
+                        r = self.obj._constructor(
+                            np.concatenate([res.values] * len(group.index)).reshape(
+                                group.shape
+                            ),
+                            columns=group.columns,
+                            index=group.index,
+                        )
+                    applied.append(r)
             else:
                 applied.append(res)
```

Comment:
can process_result return res and do this append outside the func

Reply:
One path uses `continue`, so the most immediate ways I see are to (a) use a sentinel value to indicate "do not add" or (b) move the non-adding branch outside of `process_result` (duplicating code).

Reply:
Never mind, the path using continue can _almost_ be moved ahead of any computation - it doesn't really depend on the result except for whether the result is a Series or not. I'm going to see if I can do a precursor cleanup of this method. Marking this as a draft for now.

Reply:
OK.  if the answer to my question is "no" thats OK too.